Android/app/src/main/java/app/intra/ui/IntroDialog.java

```diff
@@ -174,32 +176,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     };
 
     private final Fragment[] pages;
-    Adapter(FragmentManager fm) {
-      super(fm);
+    Adapter(Fragment fragment) {
+      super(fragment);
       pages = new Fragment[NUM_PAGES];
       for (int i = 0; i < pages.length; ++i) {
         pages[i] = Page.newInstance(images[i], headlines[i], text[i]);
       }
     }
 
     @Override
-    public int getCount() {
+    public int getItemCount() {
       return pages.length;
     }
 
     @Override
-    public Fragment getItem(int position) {
-      if (position >= 0 && position < pages.length) {
-        return pages[position];
-      }
-      return null;
+    public @NonNull Fragment createFragment(int position) {
+      return pages[position];
```

Comment:
Is `position` guaranteed to stay in bounds?

Reply:
Yes, I believe it is guaranteed to be `0 ... getItemCount()-1`.